### PR TITLE
Update tasks after build

### PR DESCRIPTION
**Pull request by Kwestor**
During each build tasks are updated for every new and/or updated file.

Fixes [#1002137](https://scala-ide-portfolio.assembla.com/spaces/scala-ide/tickets/1002137#/activity/ticket:)

Lastly, the below disclaimer is required by the lawyers:

THE FOLLOWING DISCLAIMER APPLIES TO ALL SOFTWARE CODE AND OTHER MATERIALS CONTRIBUTED IN CONNECTION WITH THIS SOFTWARE:
THIS SOFTWARE IS LICENSED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS “AS IS” AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE AND ANY WARRANTY OF NON-INFRINGEMENT, ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE. THIS SOFTWARE MAY BE REDISTRIBUTED TO OTHERS ONLY BY EFFECTIVELY USING THIS OR ANOTHER EQUIVALENT DISCLAIMER IN ADDITION TO ANY OTHER REQUIRED LICENSE TERMS.
ONLY THE SOFTWARE CODE AND OTHER MATERIALS CONTRIBUTED IN CONNECTION WITH THIS SOFTWARE, IF ANY, THAT ARE ATTACHED TO (OR OTHERWISE ACCOMPANY) THIS SUBMISSION (AND ORDINARY COURSE CONTRIBUTIONS OF FUTURES PATCHES THERETO) ARE TO BE CONSIDERED A CONTRIBUTION. NO OTHER SOFTWARE CODE OR MATERIALS ARE A CONTRIBUTION.

### CODE REVIEW

**Comment by mpociecha** [comment]
LGTM

File: org.scala-ide.sdt.core/src/org/scalaide/core/internal/builder/TaskManager.scala [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+      Comment(msg, pos) <- extractComments(iFile)
+      if pos.isDefined
+      TaskScanner.Task(tag, msg, priority, taskPos) <- taskScanner.extractTasks(msg, pos)
+      if taskPos.isDefined
+    } {
+      val start = taskPos.start
+      val length = taskPos.end - start
+      taskPos.source.file match {
+        case EclipseResource(i: IFile) =>
+          registerTask(i, tag, msg, priority, start, length, taskPos.line)
```
Comment (romanowski) [inline org.scala-ide.sdt.core/src/org/scalaide/core/internal/builder/TaskManager.scala]: wrapRegisterTask in method that takes 2 arguments i (btw rename i to some more meaningfull)

question to all: Can't we use iFile from above?

File: org.scala-ide.sdt.core/src/org/scalaide/core/internal/builder/TaskManager.scala [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+        case _ => // ignore
+      }
+    }
+  }
+
+  private def extractComments(iFile: IFile): Seq[Comment] = {
+    val scalaFile = ScalaSourceFile.createFromPath(iFile.getFullPath.toOSString)
+      .getOrElse(return Seq.empty) // TODO - handle java files?
```
Comment (romanowski) [inline org.scala-ide.sdt.core/src/org/scalaide/core/internal/builder/TaskManager.scala]: Return in by name param? I don't think it is good idea

**Comment by Kwestor** [comment]
`return` removed, some other changes (mostly renames) applied.

I also moved `clearTasks` method from `FileUtils` to `TaskManager` to keep logic for adding and removing markers in the same place.

**Comment by dragos** [comment]
I think there was a test for this functionality. Could you locate it and un-`@Ignore` it?

**Comment by dragos** [comment]
otherwise LGTM

**Comment by sschaef** [comment]
LGTM

**Comment by sschaef** [comment]
The test is `org.scalaide.core.sbtbuilder.TodoBuilderTest`.

**Comment by Kwestor** [comment]
Test re-enabled and re-written (it was kind of bleh). Do you think we can now close [SI-7363](https://issues.scala-lang.org/browse/SI-7363) now?

**Comment by Kwestor** [comment]
```
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 26:40.725s
[INFO] Finished at: Fri Apr 24 22:32:34 CEST 2015
[INFO] Final Memory: 143M/658M
[INFO] ------------------------------------------------------------------------
/tmp/hudson1043709263555576595.sh: line 44: syntax error near unexpected token `)'
/tmp/hudson1043709263555576595.sh: line 44: `    echo "An update site with the contents of this PR is available at http://download.scala-ide.org/pr-builds/scala-ide/pr-${pullrequest}/2.11.x/site")'
Build step 'Execute shell' marked build as failure
Recording test results
Finished: FAILURE
```

**Comment by sschaef** [comment]
Oh jenkins, why are you so brocken? Luckily, all tests passed. @dragos Investigating this failure is something for you.

**Comment by sschaef** [comment]
I just closed SI-7363.

**Comment by Kwestor** [comment]
Thanks!

**Comment by lrytz** [comment]
Hi @Kwestor 

cc @adriaanm @retronym

Over at https://github.com/scala/scala we are experiencing failures in the ide-integration job of our PR-validation since this commit is merged.

The failure is consistent (not flaky). The last successful build is this, which used scala-ide revision cb758fa1acb250b136c77f47314623126a1b89a1 (HEAD before merging this PR): https://scala-ci.typesafe.com/job/scala-2.11.x-integrate-ide/822/console

Since then, every build failed in the following way (if it didn't fail earlier for some other reason), here's the first one which builds scala-ide revision 6fcd0292546e5838061408309cb1841fe7d93e03 (the merge of this PR): https://scala-ci.typesafe.com/job/scala-2.11.x-integrate-ide/823/console
```
Tests in error: 
  transitive_dep_indirect(org.scalaide.core.sbtbuilder.ProjectDependenciesTest): Errors occurred during the build.
```

Sorry we didn't notify you earlier, there were other things in flux with our PR validation due to the move to bintray, and because Adriaan set up a new artifactory repo and maven proxy on our build infrastructure.

I managed to grab the logs of the failing build, i put them into a gist: https://gist.github.com/lrytz/8e31c4e1a064bf004486. Note that the stack trace of the failure (https://gist.github.com/lrytz/8e31c4e1a064bf004486#file-org-scalaide-testssuite-txt-L85) goes through the new code path introduced by this PR (https://github.com/scala-ide/scala-ide/pull/933/files#diff-44bc051d84135233f92f85c86cf7aa98R128).

**Comment by mpociecha** [comment]
@lrytz We've seen there are problems on your Jenkins (https://github.com/scala-ide/scala-ide/pull/936), but we lacked details.Thanks!

**Comment by lrytz** [comment]
Great, you were already looking into this, I didn't see that ticket!

**Comment by Kwestor** [comment]
@lrytz I don't know if there are an issue for that, but I commited a fix here: #943.

**Comment by lrytz** [comment]
@Kwestor excellent, thanks!

**Comment by dragos** [comment]
@lrytz thanks for digging this, I don't know how I missed that stack trace. I guess we can close this PR now?

**Pull request merged**